cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorNonParametrizedTest.java

```diff
@@ -148,4 +148,80 @@ public void testCustomLabelsAndAnnotations(VertxTestContext context) {
+                                .endMetadata()
+                            .endClusterCaCert()
+                        .endTemplate()
+                    .endKafka()
+                    .withNewZookeeper()
+                        .withReplicas(3)
+                        .withNewEphemeralStorage()
+                        .endEphemeralStorage()
+                    .endZookeeper()
+                .endSpec()
+                .build();
+        
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+        SecretOperator secretOps = supplier.secretOperations;
+
+        ArgumentCaptor<Secret> clusterCaCert = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clusterCaKey = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clientsCaCert = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clientsCaKey = ArgumentCaptor.forClass(Secret.class);
+        when(secretOps.reconcile(eq(NAMESPACE), eq(AbstractModel.clusterCaCertSecretName(NAME)), clusterCaCert.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(AbstractModel.clusterCaKeySecretName(NAME)), clusterCaKey.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(KafkaCluster.clientsCaCertSecretName(NAME)), clientsCaCert.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(KafkaCluster.clientsCaKeySecretName(NAME)), clientsCaKey.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+
+        KafkaAssemblyOperator op = new KafkaAssemblyOperator(vertx, new PlatformFeaturesAvailability(false, KubernetesVersion.V1_9), certManager, passwordGenerator,
+                supplier, ResourceUtils.dummyClusterOperatorConfig(1L));
+        Reconciliation reconciliation = new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, NAME);
+
+        Checkpoint async = context.checkpoint();
+        
+        op.new ReconciliationState(reconciliation, kafka).reconcileCas(() -> new Date())
+                .onComplete(context.succeeding(c -> context.verify(() -> {
+                    assertThat(clusterCaCert.getAllValues(), hasSize(1));
+                    assertThat(clusterCaKey.getAllValues(), hasSize(1));
+                    assertThat(clientsCaCert.getAllValues(), hasSize(1));
+                    assertThat(clientsCaKey.getAllValues(), hasSize(1));
+
+                    Secret clusterCaCertSecret = clusterCaCert.getValue();
+                    Secret clusterCaKeySecret = clusterCaKey.getValue();
+                    Secret clientsCaCertSecret = clientsCaCert.getValue();
+                    Secret clientsCaKeySecret = clientsCaKey.getValue();
```

Comment:
Can you check the owner reference for all of the secrets? Although your PR affects only the cert secrets and not for the key secrets, it makes sense to test it to make sure they are not affected.

Reply:
Sure!